datumaro/plugins/data_formats/arrow/exporter.py

```diff
@@ -0,0 +1,418 @@
+        else:
+            setattr(self, "__writer_idx", (idx + 1) % self._num_shards)
+
+        if len(self._writers) <= idx:
+            writer, fname = self._init_writer(idx)
+            self._writers.append(writer)
+            self._fnames.append(fname)
+        self._writers[idx].write_table(pa_table, max_chunk_size)
+
+    def write(self, max_chunk_size: Optional[int] = None, pool: Optional[Pool] = None):
+        if max_chunk_size is None:
+            max_chunk_size = self._max_chunk_size
+
+        if len(self.items) < max_chunk_size:
+            return
+
+        batch = [[] for _ in self._schema.names]
+        for item in self._ctx.progress_reporter.iter(
+            self.items, desc=f"Building arrow for {self._subset}"
+        ):
+            if isinstance(item, ApplyResult):
+                item = item.get(timeout=DatumaroArrow.MP_TIMEOUT)
+            if isinstance(item, partial):
+                item = item()
+            for j, name in enumerate(self._schema.names):
+                batch[j].append(item[name])
+
+            if len(batch[0]) >= max_chunk_size:
+                self._write(batch, max_chunk_size)
+                batch = [[] for _ in self._schema.names]
+        if len(batch[0]) > 0:
+            self._write(batch, max_chunk_size)
+
+        self._data["items"] = []
+
+    def done(self):
+        total = len(self._fnames)
+        width = len(str(total))
+        for idx, (fname, writer) in enumerate(zip(self._fnames, self._writers)):
+            writer.close()
+            if platform.system() != "Windows":
+                template = PathNormalizer.unnormalize(fname)
+                placeholders = {"width": width, "total": total, f"idx{idx}": idx}
+                new_fname = template.format(**placeholders)
+                if os.path.exists(new_fname):
+                    os.remove(new_fname)
+                os.rename(fname, new_fname)
+
+
+class ArrowExporter(Exporter):
+    AVAILABLE_IMAGE_EXTS = ImageMapper.AVAILABLE_SCHEMES
+    DEFAULT_IMAGE_EXT = ImageMapper.AVAILABLE_SCHEMES[0]
+
+    @classmethod
+    def build_cmdline_parser(cls, **kwargs):
+        parser = super().build_cmdline_parser(**kwargs)
+
+        # '--image-ext' would be used in a different way for arrow foramt
+        _actions = []
+        for action in parser._actions:
+            if action.dest != "image_ext":
+                _actions.append(action)
+        parser._actions = _actions
+        parser._option_string_actions.pop("--image-ext")
+
+        parser.add_argument(
+            "--image-ext",
+            default=None,
+            help=f"Image encoding scheme. (default: {cls.DEFAULT_IMAGE_EXT})",
+            choices=cls.AVAILABLE_IMAGE_EXTS,
+        )
+
+        #  parser.add_argument(
+        #      "--split-by-subsets",
+        #      action="store_true",
+        #      default=False,
+        #      help="Split arrow dataset by subset. (default: %(default)s)",
+        #  )
```

Comment:
Can we remove?

Reply:
Fixed in https://github.com/openvinotoolkit/datumaro/pull/931/commits/0a8eda950dd54f0b83c2762fb1731f7a61bdc0e9